### PR TITLE
Added tests for the convolve with the memoryviews in the docs.

### PR DESCRIPTION
**Pull request by gabrieldemarmiesse**
I added tests for the examples directory. For the moment it only covers the memoryview tests. 

I would like some feedback on the approach and if it's good, I'll enable more tests for files in the examples directory.

I added a else close when checking the value of `mode` because I think it'll more clear for people who are getting into the `runtests.py` (I know I wanted a compile test, so I looked for the string 'compile' and it took me a while to figure out where the mode was tested).

### CODE REVIEW

File: runtests.py [hunk 1]
```diff
@@ -610,8 +614,10 @@ def handle_directory(self, path, context):
                     test_class = CythonUnitTestCase
                 else:
                     test_class = CythonRunTestCase
-            else:
+            elif mode in ['compile', 'error', 'test']:
```
Comment (scoder) [inline runtests.py]: Is there really a 'test' test mode?

File: runtests.py [hunk 2]
```diff
@@ -554,7 +558,7 @@ def build_suite(self):
                     continue
                 suite.addTest(
                     self.handle_directory(path, filename))
-        if sys.platform not in ['win32']:
+        if sys.platform not in ['win32'] and self.add_embedded_test:
```
Comment (scoder) [inline runtests.py]: Hmm, this embed-test integration really is a hack. And excluding it with this flag makes it even worse. If anything, then _including_ the test should be done explicitly, not _excluding_ it.

File: runtests.py [hunk 3]
```diff
@@ -2130,6 +2142,18 @@ def runtests(options, cmd_args, coverage=None):
...
+        filetests = TestBuilder(options.examples_dir, WORKDIR, selectors, exclude_selectors,
+                                options.annotate_source, options.cleanup_workdir,
+                                options.cleanup_sharedlibs, options.cleanup_failures,
+                                options.pyregr,
+                                options.cython_only, languages, test_bugs,
+                                options.fork, options.language_level,
+                                options.test_determinism,
+                                common_utility_dir, options.pythran_dir,
+                                default_mode='compile',
+                                add_embedded_test=False)
```
Comment (scoder) [inline runtests.py]: Hmm, I think seeing all these options makes it clear that this needs some kind of refactoring... Not sure which yet, but this doesn't look right...

File: runtests.py (hunk 1 above)
Comment (gabrieldemarmiesse) [reply runtests.py]: Yes, you can see it on the first failed build of this PR

File: runtests.py (hunk 3 above)
Comment (gabrieldemarmiesse) [reply runtests.py]: Passing the option variable? Then extract the members in the __init__()?

File: runtests.py (hunk 2 above)
Comment (gabrieldemarmiesse) [reply runtests.py]: Done

**Comment by gabrieldemarmiesse** [comment]
Does this refactoring suits you? If not, the easiest would be for you to give me the API that you have in mind, or that you do the refactoring on the master branch and I'll adapt my PR to it.

**Comment by gabrieldemarmiesse** [comment]
I am not entirely satisfied by this approach because I had to remove the
```
# distutils: sources=./C_func_file.c
```
 in favor of 
```
cdef extern from "C_func_file.c":
    # C is include here so that it doesn't need to be compiled externally
    pass
```
because I needed the path of the C file relative **to the pyx file** and that's not possible with a compiler directive.
I used this method because it was used in `Cython/tests/run/cpp_function_lib.pxd` test case.

File: docs/examples/memoryviews/memview_to_c.pyx [hunk 4]
```diff
@@ -1,5 +1,6 @@
-# distutils: sources=./C_func_file.c
-# distutils: include_dirs=./
+cdef extern from "C_func_file.c":
+    # C is include here so that it doesn't need to be compiled externally
```
Comment (scoder) [inline docs/examples/memoryviews/memview_to_c.pyx]: I agree that it's not beautiful, but I think it's ok.

**Pull request merged**